### PR TITLE
Dibyasom | Regression/Simple_Linear_Regression | Implemented with python3 from scratch.

### PR DESCRIPTION
**Pull request by vhaegar1526**
Non SKLearn based solution, with matplotlib visualizer.

### CODE REVIEW

File: 3_Regression/Simple_Linear_Regression/python/python-scratch/LinearRegression_scratch.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
+# Linear Regression Model.
```
Comment (NishkarshRaj) [inline 3_Regression/Simple_Linear_Regression/python/python-scratch/LinearRegression_scratch.py:1]: You are clearly contributing someone else's code?

**Review by NishkarshRaj — CHANGES_REQUESTED** [review]
Explain the comment given by me in the codes.

File: 3_Regression/Simple_Linear_Regression/python/python-scratch/LinearRegression_scratch.py (hunk 1 above)
Comment (vhaegar1526) [reply 3_Regression/Simple_Linear_Regression/python/python-scratch/LinearRegression_scratch.py:1]: Sir it's mine, i'm 2nd year student and my domain is AIML.
Specifically, it's one of my assignment from 2nd semester.

**Comment by vhaegar1526** [comment]
Sir, with context your code, explaining the #comments.

# Importing the libraries
Importing the necessary libraries, viz..
*Matplotlib for plotting the regression line and data points.
*Pandas for handling and manipulating csv-data file.
*SkLearn for data preprocessing and implementing linear-regression using sklearn-linear model.

# Importing the dataset
Fetches the dataset from parent node.

# Splitting the dataset into the Training set and Test set
Splits the dataset using sklearn library, into 66.66% train split and 33.33% for test.

 # Fitting Simple Linear Regression to the Training set
Feeds the tuples in train split into the predefined linear model in sklearn.

# Predicting the Test set results
Uses predict(x) method of linear model, to generate Y_pred.
Basically, fits the X_test into the line eq of linear-regression parameters.(Slope and intercept)

# Visualising the Training set results
# Visualising the Test set results
Separately plots the train and test data points in different subplots.

Thankyou.

**Comment by NishkarshRaj** [comment]
> 
> 
> Sir, with context to your code, explaining the #comments.
> # Importing the libraries
> 
> Importing the necessary libraries, viz..
> *Matplotlib for plotting the regression line and data points.
> *Pandas for handling and manipulating csv-data file.
> *SkLearn for data preprocessing and implementing linear-regression using sklearn-linear model.
> # Importing the dataset
> 
> Fetches the dataset from the parent node.
> # Splitting the dataset into the Training set and Test set
> 
> Splits the dataset using sklearn library, into 66.66% train split and 33.33% for test.
> # Fitting Simple Linear Regression to the Training set
> 
> Feeds the tuples in train split into the predefined linear model in sklearn.
> # Predicting the Test set results
> 
> Uses predict(x) method of linear model, to generate Y_pred.
> Basically, fits the X_test into the line eq of linear-regression parameters.(Slope and intercept)
> # Visualising the Training set results
> # Visualising the Test set results
> 
> Separately plots the train and test data points in different subplots.
> 
> Thankyou.

@vhaegar1526 I know you are knowledgeable and bright. I am not stating you don't know stuff.

The code you submitted had the name of the author and your profile no where states that name is yours! You should have mentioned that in the PR or use your Real name in GitHub Profile.

Thanks for the clarification though 😄

**Review by NishkarshRaj — APPROVED** [review]

**Comment by NishkarshRaj** [comment]
Thanks @vhaegar1526 for your contribution 👍

**Comment by vhaegar1526** [comment]
Sorry for the confusion, I did miss mentioning that. I'm Dibyasom, I've sent you a linked in connection request too.
